src/engraving/editing/cmd.cpp

```diff
@@ -3375,6 +3374,108 @@ void Score::cmdIncDecDuration(int nSteps, bool stepDotted)
     }
 }
 
+//---------------------------------------------------------
+//   extendToNextNote
+//---------------------------------------------------------
+
+void Score::cmdExtendToNextNote()
+{
+    const Fraction startTick = selection().tickStart();
+    Fraction endTick = selection().tickEnd();
+    const staff_idx_t startStaff = selection().staffStart();
+    const staff_idx_t endStaff = selection().staffEnd();
+
+    std::vector<ChordRest*> toSelect;
+    const bool wasRangeSelection = selection().isRange();
+
+    for (EngravingItem* el : selection().elements()) {
+        if (!el->isNote()) {
+            continue;
+        }
+        Note* n = toNote(el);
+        ChordRest* cr = toChordRest(n->chord());
+        toSelect.push_back(cr);
+        InputState& is = score()->inputState();
+
+        if (cr->endTick() == lastSegment()->tick()) {
+            continue;
+        }
+
+        // fill next empty measure with rest for voices > 0
+        if (cr->voice() != 0 && !nextChordRest(cr)) {
+            Measure* m = cr->measure()->nextMeasure();
+            is.setTrack(cr->track());
+            is.setSegment(m->segments().firstCRSegment());
+            score()->enterRest(m->ticks());
+        }
+
+        ChordRest* ncr = nextChordRest(cr);
+
+        while (ncr->isRest() || (ncr->tick() > cr->endTick())) { //second condition checks for empty measure between cr and ncr for voices 2,3,4.
+            if (cr->endTick() != ncr->tick()) { // check for empty measures and fill with mRest for voices>0
+                is.setTrack(cr->track());
+                is.setSegment(cr->measure()->nextMeasure()->segments().firstCRSegment());
+                score()->enterRest(cr->measure()->nextMeasure()->ticks());
```

Comment:
Instead of first entering a rest and then immediately replacing that rest, it would be nicer to extend the chord directly. Or would that be complicated?

Reply:
Would it be an idea to put the block in a static function or a lambda function and call that twice?